### ISSUE TITLE
 AIP-121 Linter does not verify uniformness of resource across methods

### ISSUE THREAD

**Issue by apodznoev** [googleapis/api-linter#1374]
Hello!

As of [AIP-121](https://google.aip.dev/121#methods) definition for resource-oriented design, 

> If the request to or the response from a standard method (or a custom method in the same service) is the resource or contains the resource, the resource schema for that resource across all methods **must** be the same.

However, having a mixed service defined like:

```
service BookService {
    rpc CreateBook(CreateBookRequest) returns (Book);

    rpc GetBook(GetBookRequest) returns (Book);

    rpc ListBooks(ListBooksRequest) returns (ListBooksResponse){
        option (google.api.method_signature) = "parent";
    }

    rpc CreateFoo(CreateFooRequest) returns (Foo);

    rpc GetFoo(GetFooRequest) returns (Foo);

    rpc ListFoos(ListFoosRequest) returns (ListFoosResponse){
        option (google.api.method_signature) = "parent";
    }
}
```
will be totally valid yielding no issues from AIP linter in regards to AIP-121 rules.

Expectation would be, that AIP reject such a service as violating same-resource schema.

I attached the full service definition which didn't produce any warning from AIP linter as of version **v1.65.1**.
[book_service.txt](https://github.com/googleapis/api-linter/files/14981856/book_service.txt)

**Comment by noahdietz** [comment]
Thanks for the issue!

If I'm understanding what you are suggesting, your example proto should be non-compliant b.c a single proto service contains RPCs for two different resources? If I'm tracking...

...then the proto you shared isn't actually a problem - AIP-121 isn't opinionated about RPCs being defined in Resource-specific proto services. That is to say, one proto service can contain more than one resource, and the methods for them.

**Comment by noahdietz** [comment]
I'm going to close this, but if I'm misunderstanding, please say as much and we can reopen.
